trees.c

```
@@ -708,6 +708,25 @@ void Z_INTERNAL zng_tr_flush_block(deflate_state *s, unsigned char *buf, uint32_
 /* ===========================================================================
  * Send the block data compressed using the given Huffman trees
  */
+/* One 64-bit little-endian load at sx spans the next three symbols' dist fields
+ * (bits 0..15, 24..39, 48..63) and two of their literals (bits 16..23, 40..47). */
+#define LIT3_DIST_MASK 0xFFFF00FFFF00FFFFULL
+
+/* The dist fields start at bit 0, 24 and 48, so the lowest set bit of the masked word
+ * says which one broke the literal run: ctz/24 is the number of literals that follow,
+ * and no bits set at all means three. */
+Z_FORCEINLINE static unsigned lit_run(uint64_t v) {
```

Cppcheck (2.21.0)

[style] The function 'lit_run' is never used.

(unusedFunction)